conans/test/integration/toolchains/cmake/test_cmaketoolchain.py

```diff
@@ -1575,3 +1574,33 @@ def _assert_install(out):
     c.run("build .")
     _assert_install(c.out)
     assert "CMAKE_INSTALL_PREFIX" not in c.out
+
+
+def test_toolchain_extra_variables():
+    windows_profile = textwrap.dedent("""
+        [settings]
+        os=Windows
+        arch=x86_64
+        [conf]
+        tools.cmake.cmaketoolchain:extra_variables={'CMAKE_GENERATOR_INSTANCE': '"${GENERATOR_INSTANCE}/buildTools/"', 'FOO': '42 CACHE' }
```

Comment:
I like the feature, but I'm not too sure about defining cache variables this way?  having the spaces be separators for cmake syntax  I'm not too sure.


```
set(FOO 42 CACHE)
```

IIRC, this is not valid CMake syntax:
```
CMake Error at CMakeLists.txt:5 (set):
  set given invalid arguments for CACHE mode.

```
(I think the type and dosctring are missing, see [docs](https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry))

I wonder if we could have something like what CMake uses for [presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html):

```
      "cacheVariables": {
        "FIRST_CACHE_VARIABLE": {
          "type": "BOOL",
          "value": "OFF"
        },
        "SECOND_CACHE_VARIABLE": "ON"
      },
```

we could maybe have something like this, where we either take the value (which shouldn't allow spaces unless they are inside double quotes?) or a dictionary with the value, whether it is cache or not, and if it is cache, the other needed info (like the type)
```
tools.cmake.cmaketoolchain:extra_variables={'CMAKE_GENERATOR_INSTANCE': '"${GENERATOR_INSTANCE}/buildTools/"', 'FOO': {value:'42', cache: true, type: 'bool'}} }
```

Reply:
Alright, take a look at the new design. It is now fully working and tested. 
Now we are supporting:
```
{
'A': 'string value',
'B': 42, 
'C': 1.3,
'DICT': {'value': '42 sense', 'cache': true, 'type': 'BOOL', 'docstring': 'test cache var'} 
}
```
Dict values includes:
* Support for pythonic boolean inputs or string inputs (in `cache` variable)
* Descriptive error if cache variable and not `type` or `docstring` defined
* Descriptive error if cache type not in `BOOL, FILEPATH, PATH, STRING, INTERNAL` list